src/hooks/index.ts

```diff
@@ -4,3 +4,9 @@ export { default as useFetch } from './use-fetch';
 export { default as useWaitForFonts } from './use-wait-for-fonts';
 export { default as useReactRoot } from './use-react-root';
 export { default as useExtendedTheme } from './use-extended-theme';
+export {
+  default as useMeasureText,
+  type UseMeasureTextProps,
```

Comment:
Might want to export types from a single location instead? Ex. `src/types` @a-m-dev

Reply:
if you create a type file within `use-measure-text` folder (which will [export all types](https://github.com/qlik-oss/nebula-table-utils/blob/main/src/components/HeadCellMenu/types.ts) within the scope of that hook + export everything from that type [file from index file of use-measure-text](https://github.com/qlik-oss/nebula-table-utils/blob/e411cfca185820bb641a7ec3c0f91dbb9e0c4ff2/src/components/HeadCellMenu/index.ts#L4)) it will be bundled and also will be accessible through:
```tsx
import useMeasureText, { your, types, here } from '@qlik/nebula-table-utils/lib/hooks/use-measure-text';
```

Currently this is the case for [HeadCellMenu component as well](https://github.com/qlik-oss/sn-pivot-table/blob/b9eb90cced7f19613136884bc0d00847a56574e3/src/pivot-table/components/cells/DimensionTitleCell.tsx#L3) (but it only exports the flag types from index file, in your case it will be `export * from './types'`)

The reason behind it:
if you **only** wants to use the component -> you have two solutions:
1. pull it from `.../lib/components` (which is the best case for only component use)
2. be more explicit and pull it from index file of the component directory `.../lib/componsnts/MYComponent` (the optional way of importing component, but it has a purpose why it exists 👇)

In your case (+ head cell menu case as well) since we need some extra information from that component (take more types info for example) we have to be more explicit and call it directly from where it is located. 

**Alternate:**
Of course you can import component from `.../lib/components` and for it's required types you can be more explicit and retrieve them from components index file, but that is optional (why have 2 lines of import when you can have one -> like HeadCellMenu import in PVT)

So having a global `src/types` is not a good idea since in terms of packaging (IMO) because then -> it might include all types from different modules/components/utils and it will ignore the scope of each of them.